### DISCUSSION TITLE
How to use SSL under Linux (Debian)?

### DISCUSSION THREAD

**Discussion by Einhornyordle** [jchristn/WatsonWebsocket#59] [General]
I created a simple WatsonWsServer and set the ssl bool to true but how do i specify the ssl certificate to use under debian (since there is no netsh command available)?

**Comment by jchristn** [comment]
Hi @Einhornyordle I unfortunately don't have a good example for you.  I'd suggest looking into ways of binding ```HttpListener``` with a certificate in either Mono or .NET Core Linux environments.  The links I'm seeing don't look too promising in terms of native SSL support with ```HttpListener``` (which WatsonWebsocket is based on).  For example:

- https://github.com/dotnet/runtime/issues/27391 - specifically
```
karelz commented on Oct 22, 2018 • 
@martin-frydl I think we're running in circles here.
.NET Core does not support SSL in HttpListener on Linux. Using Reflection to hack-enable features is unsupported and we want to DISCOURAGE anyone from doing that. Fixing bugs on such code paths proactively is encouragement and goes against our goals. Hence, the priority is extremely low. We did not close the issue right away just because it will be needed one day for full SSL support in HttpListener on Linux (#19752).
```
- https://github.com/dotnet/runtime/issues/19752

This may or may not be helpful (it is for Mono environments, but may work for .NET Core): http://joshua.perina.com/geo/post/using-ssl-https-with-mono-httplistener

If not, you may have to run your code behind something like nginx. 

Would you mind trying the link and let us know how it goes?

**Reply by sungaila** [reply]
I've got SSL running on my Ubuntu Server. The .NET app is running without SSL and listens to localhost:3001. Nginx is used as reverse proxy to redirect wss://example.com:3000 to ws://localhost:3001.

This StackOverflow post has the correct nginx configuration: https://stackoverflow.com/a/14969925

Do not put `proxy_set_header X-Real-IP $remote_addr;` and the other stuff into the config. It will result in a 404 not found.

Also don't forget to add the SSL stuff into the nginx config.
